registry/manifest.go

```diff
@@ -2,44 +2,26 @@ package registry
 
 import (
 	"bytes"
-	"encoding/json"
 	"io"
 	"net/http"
 
-	"github.com/docker/distribution/manifest/ocischema"
+	"github.com/docker/distribution/manifest/manifestlist"
+	ociV1 "github.com/docker/distribution/manifest/ocischema"
 	manifestV1 "github.com/docker/distribution/manifest/schema1"
 	manifestV2 "github.com/docker/distribution/manifest/schema2"
 	"github.com/opencontainers/go-digest"
-	ociSpec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
+// Opt to define these constants here instead of importing
+// github.com/opencontainers/image-spec/specs-go/v1
+// to ensure we use the docker/distribution library for unmarshalling purposes.
 const (
-	MediaTypeManifestList = "application/vnd.docker.distribution.manifest.list.v2+json"
+	// MediaTypeImageManifest specifies the media type for an image manifest.
+	MediaTypeImageManifest = "application/vnd.oci.image.manifest.v1+json"
+	// MediaTypeImageIndex specifies the media type for an image index.
+	MediaTypeImageIndex = "application/vnd.oci.image.index.v1+json"
```

Comment:
Curious, why decided to remove the `ociSpec` import and create these `const`'s instead of using `ociSpec.MediaTypeImageManifest` and `ociSpec.MediaTypeImageIndex`?

Reply:
commented above:
```
// Opt to define these constants here instead of importing
// github.com/opencontainers/image-spec/specs-go/v1
// to ensure we use the docker/distribution library for unmarshalling purposes.
```

Combination of that ^ plus also wanted to reduce imports/dependencies (though I can see that library is still an indirect dependency)

Reply:
To build on the stuff in the comments, we had already been using `docker/distribution` for unmarshalling the manifests, and it already implemented ways for us to unmarshall OCI-compliant media-types, too, so I figured it'd be better to just stick with that than mix in the extra dependency